src/Datadog.Trace/Configuration/JsonConfigurationSource.cs

```diff
@@ -118,8 +118,34 @@ public T GetValue<T>(string key)
         /// </remarks>
         /// <param name="key">The key that identifies the setting.</param>
         /// <returns><see cref="IDictionary{TKey, TValue}"/> containing all of the key-value pairs.</returns>
-        /// <exception cref="JsonReaderException">Thrown if the configuration value is not a valid JSON string.</exception>"
+        /// <exception cref="JsonReaderException">Thrown if the configuration value is not a valid JSON string.</exception>
         public IDictionary<string, string> GetDictionary(string key)
+        {
+            return GetDictionaryInternal(key, false);
```

Comment:
```suggestion
            return GetDictionaryInternal(key, allowOptionalMappings: false);
```

Reply:
Done in c1ac5ce1868b7c359e741bd04d05e742647a9a33